FRAuth/FRAuth/DeviceBinding/DeviceBindingAuthenticators.swift

```diff
@@ -114,180 +125,222 @@ extension DeviceAuthenticator {
     /// local authentication policy for authentication
     var policy: LAPolicy
-    /// keyAware for key pair generation
-    var keyAware: KeyAware
     
     
-    /// Initializes BiometricOnly with given BiometricHandler and KeyAware
-    /// - Parameter biometricInterface: biometric handler for authentication
-    /// - Parameter keyAware: keyAware for key pair generation
-    init(description: String, keyAware: KeyAware) {
-        self.promptDescription = description
-        self.keyAware = keyAware
+    /// Initializes BiometricOnly with the rightLAPolicy
+    override init() {
         policy = .deviceOwnerAuthentication
     }
     
     
     /// Generate public and private key pair
-    func generateKeys() throws -> KeyPair {
-        var keyBuilderQuery = keyAware.keyBuilderQuery()
+    open func generateKeys() throws -> KeyPair {
+        guard let cryptoKey = cryptoKey, let prompt = prompt else {
+            throw DeviceBindingStatus.unsupported(errorMessage: "Cannot generate keys, missing cryptoKey or prompt")
+        }
+        
+        var keyBuilderQuery = cryptoKey.keyBuilderQuery()
         keyBuilderQuery[String(kSecAttrAccessControl)] = accessControl()
         
 #if !targetEnvironment(simulator)
         let context = LAContext()
-        context.localizedReason = promptDescription
+        context.localizedReason = prompt.description
         keyBuilderQuery[String(kSecUseAuthenticationContext)] = context
 #endif
         
-        return try keyAware.createKeyPair(builderQuery: keyBuilderQuery)
+        return try cryptoKey.createKeyPair(builderQuery: keyBuilderQuery)
     }
     
     
     /// Check if authentication is supported
-    func isSupported() -> Bool {
+    open func isSupported() -> Bool {
         let laContext = LAContext()
         var evalError: NSError?
         return laContext.canEvaluatePolicy(policy, error: &evalError)
     }
     
     
     /// Access Control for the authetication type
-    func accessControl() -> SecAccessControl? {
+    open func accessControl() -> SecAccessControl? {
 #if !targetEnvironment(simulator)
         return SecAccessControlCreateWithFlags(kCFAllocatorDefault, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, [.userPresence, .privateKeyUsage], nil)
 #else
         return SecAccessControlCreateWithFlags(kCFAllocatorDefault, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, [.userPresence], nil)
 #endif
     }
-}
-
-
-internal class None: DeviceAuthenticator {
-    
-    /// keyAware for key pair generation
-    var keyAware: KeyAware
     
     
-    /// Initializes BiometricOnly with given BiometricHandler and KeyAware
-    /// - Parameter keyAware: keyAware for key pair generation
-    init(keyAware: KeyAware) {
-        self.keyAware = keyAware
+    open func type() -> DeviceBindingAuthenticationType {
+        return .biometricAllowFallback
     }
+}
+
+
+open class None: DeviceAuthenticator, CryptoAware {
```

Comment:
Super minor but not a major fan of this name. Can we get something more descriptive?

Reply:
We have the same name for Android. Yes it's a bit generic I agree, but it's aligned with Android naming. @jeyanthanperiyasamy what do you think, should we rename it?

Reply:
@vahancouver  @george-bafaloukas-forgerock  . this is resembling the Server Authentication type, yeah its not a great name , otherwise we should also give a better name in server as well.  i am ok to leave as it is.